desktop/cmp/dash/canvas/DashCanvas.js

```diff
@@ -26,48 +29,58 @@ export const [DashCanvas, dashCanvas] = hoistCmp.withFactory({
     model: uses(DashCanvasModel),
     render({className, model}) {
         const isDraggable = !model.layoutLocked,
-            isResizable = !model.layoutLocked;
+            isResizable = !model.layoutLocked,
+            {ref} = model;
 
         return div({
             className: classNames(
                 className,
                 isDraggable ? `${className}--draggable` : null,
                 isResizable ? `${className}--resizable` : null
             ),
-            ref: model.ref,
+            ref,
             onContextMenu: (e) => {
-                const {clientX: x, clientY: y} = e;
-                ContextMenu.show(
-                    dashCanvasContextMenu({
-                        dashCanvasModel: model,
-                        clickPosition: {x, y}
-                    }),
-                    {left: x, top: y},
-                    null,
-                    XH.darkTheme
-                );
+                const {clientX, clientY, target} = e,
```

Comment:
put the variables and computation you need for the computation within the if block.